metrics-operator/controllers/common/providers/datadog/datadog_test.go

```diff
@@ -410,6 +462,461 @@ func TestEvaluateQuery_WrongInterval(t *testing.T) {
 	require.Empty(t, r)
 }
 
+func TestEvaluateQueryForStep_APIError(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte(ddErrorPayload))
+		require.Nil(t, err)
+	}))
+	defer svr.Close()
+
+	secretName := "datadogSecret"
+	apiKey, apiKeyValue := "DD_CLIENT_API_KEY", "fake-api-key"
+	appKey, appKeyValue := "DD_CLIENT_APP_KEY", "fake-app-key"
+	apiToken := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: "",
+		},
+		Data: map[string][]byte{
+			apiKey: []byte(apiKeyValue),
+			appKey: []byte(appKeyValue),
+		},
+	}
+	kdd := setupTest(apiToken)
+	metric := metricsapi.KeptnMetric{
+		Spec: metricsapi.KeptnMetricSpec{
+			Query: "system.cpu.idle{*}",
+			Range: &metricsapi.RangeSpec{
+				Interval:    "5m",
+				Step:        "1m",
+				Aggregation: "max",
+			},
+		},
+	}
+	b := true
+	p := metricsapi.KeptnMetricsProvider{
+		Spec: metricsapi.KeptnMetricsProviderSpec{
+			SecretKeyRef: v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: secretName,
+				},
+				Optional: &b,
+			},
+			TargetServer: svr.URL,
+		},
+	}
+	r, raw, e := kdd.EvaluateQueryForStep(context.TODO(), metric, p)
+	require.Error(t, e)
+	require.Contains(t, e.Error(), "Token is missing required scope")
+	require.Equal(t, []byte(ddErrorPayload), raw)
+	require.Empty(t, r)
+}
+
+func TestEvaluateQueryForStep_HappyPath(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte(ddPayload))
+		require.Nil(t, err)
```

Comment:
Let's also add some checks for the several query parameters to be submitted as we expect them, i.e. something like:

```
require.Equal(t, "1m", r.URL.Query().Get("interval"))
```

Reply:
Just added this for this particular test case, does it needs to be get added to other test cases as well or only for this one?

Also, the interval would be equal to `60000` as we convert `1m` to milliseconds.
```
require.Equal(t, "60000", r.URL.Query().Get("interval"))
```

Reply:
Thanks, I think it's sufficient to have the checks for this test case